src/time-picker/TimePicker.js

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import { props, t, skinnable } from '../utils';
+import cx from 'classnames';
+import Dropdown from '../dropdown/Dropdown';
+import range from 'lodash/utility/range';
+import flatten from 'lodash/array/flatten';
+
+/*
+  Options is defined here cause we need to not change the ref.
+  This happend because of a very wired behaviour of react-select v0.6.12
+*/
+const options = [];
+
+const H24 = '24h';
+const H12 = '12h';
+const symbolRegex = /[\s,\.\|\/\\]/;
+const numberRegex = /^\d+$/;
+const time12Regex = /^([1-9]|1[0-2])[:,\.\|\/\\]?([0-5]\d)?\s*?(AM|PM)$/;
+const time24Regex = /^([01]?\d|2[0-3])[:,\.\|\/\\]?([0-5]?\d)?$/;
+const separator = ':';
+
+const pad = (num) => num <= 9 ? `0${num}` : num;
+const timeFormatter24 = (hour, minute) => `${pad(hour)}:${pad(minute)}`;
+const timeFormatter12 = (hour, minute) => {
+  if (hour === 0 || (hour > 12 && hour % 12 === 0)) {
+    return `12${separator}${pad(minute)} am`;
+  } else if (hour === 12) {
+    return `12${separator}${pad(minute)} pm`;
+  } else if (hour >= 13) {
+    return `${pad(hour % 12)}${separator}${pad(minute)} pm`;
+  } else {
+    return `${pad(hour)}${separator}${pad(minute)} am`;
+  }
+};
+
+const isValidTime = input => time12Regex.test(input) || time24Regex.test(input);
+
+const formatValue = (timeFormatter, hour, minute) => ({
+  value: `${hour}${separator}${minute}`,
+  label: timeFormatter(hour, minute)
+});
+
+const lteTime = (minTime, maxTime) => (
+  maxTime.hours > minTime.hours || (maxTime.hours === minTime.hours && maxTime.minutes >= minTime.minutes)
+);
+
+const addZero = str => {
+  if (str.length === 4) {
+    return `${str}0`;
+  } else {
+    return str;
+  }
+};
+
+const insertColon = str => {
```

Comment:
`insertColon` (where the name have to be refactored into `insertSeparator` IMHO) it's useful to handle another use case wanted by the requirements, so when the input have no separator. It will insert the separator (by default `:`) right in the middle of hours and minute. The code of this function is not very clear i guess, i'm starting implementing this from a code snippet i found. It can be refactored.